code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -12,28 +12,84 @@
 	max_heat_protection_temperature = SPACE_SUIT_MAX_TEMP_PROTECT
 	species_restricted = list("Plasmaman")
 	flags = STOPSPRESSUREDMAGE
+	autoextinguish = 1
 
 	icon_state = "plasmaman_suit"
 	item_state = "plasmaman_suit"
 
+	actions_types = list(/datum/action/item_action/eject_depleted_cartridges)
```

Comment:
This is ridiculously overcomplicated. I'd just go with TG's system of being based on uses (starts off with 5) with using a cartridge on the suit resetting uses to 5....it'll save this needless over-complexity.